clients/venice-push-job/src/test/java/com/linkedin/venice/hadoop/VenicePushJobTest.java

```diff
@@ -877,6 +881,49 @@ public void testGetPerColoPushJobDetailsStatusFromExecutionStatus() {
     }
   }
 
+  /**
+   * Tests that the error message for the {@link VenicePushJob.PushJobCheckpoints#RECORD_TOO_LARGE_FAILED} code path of
+   * {@link VenicePushJob#updatePushJobDetailsWithJobDetails(DataWriterTaskTracker)} uses maxRecordSizeBytes.
+   */
+  @Test(dataProvider = "True-and-False", dataProviderClass = DataProviderUtils.class)
+  public void testUpdatePushJobDetailsWithJobDetailsRecordTooLarge(boolean chunkingEnabled) {
+    try (final VenicePushJob vpj = getSpyVenicePushJob(getVpjRequiredProperties(), getClient())) {
+      // Setup push job settings and mocks
+      PushJobDetails pushJobDetails = vpj.getPushJobDetails();
+      pushJobDetails.chunkingEnabled = chunkingEnabled;
+      PushJobSetting pushJobSetting = vpj.getPushJobSetting();
+      setPushJobSettingDefaults(pushJobSetting);
+      vpj.setInputStorageQuotaTracker(mock(InputStorageQuotaTracker.class));
+      final DataWriterTaskTracker dataWriterTaskTracker = mock(DataWriterTaskTracker.class);
+      doReturn(1L).when(dataWriterTaskTracker).getRecordTooLargeFailureCount();
+
+      // The value of chunkingEnabled should dictate the error message returned
+      final String errorMessage = vpj.updatePushJobDetailsWithJobDetails(dataWriterTaskTracker);
+      final int latestCheckpoint = pushJobDetails.pushJobLatestCheckpoint;
+      Assert.assertTrue(errorMessage.contains((chunkingEnabled) ? "Venice" : "Kafka"), errorMessage);
+      Assert.assertTrue(errorMessage.contains((chunkingEnabled) ? "10.0 MiB" : "950.0 KiB"), errorMessage);
```

Comment:
Instead of splitting this across two asserts, can we check once for the composed error message? It will help with readability

Reply:
Irrelevant since [the system name was removed](https://github.com/linkedin/venice/pull/1052#discussion_r1669385417). I originally checked for the full error message, but I felt that it was slightly harder to read because it was so long.